app/src/androidTest/java/ch/epfl/qedit/answer/MatrixFragmentTest.java

```diff
@@ -8,39 +8,54 @@
 import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
 import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static ch.epfl.qedit.view.quiz.QuestionFragment.ANSWER_FORMAT;
+import static ch.epfl.qedit.view.quiz.QuestionFragment.ANSWER_MODEL;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertNull;
 
 import android.os.Bundle;
+import androidx.lifecycle.ViewModelProvider;
 import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;
 import ch.epfl.qedit.R;
 import ch.epfl.qedit.backend.database.DatabaseFactory;
 import ch.epfl.qedit.backend.database.MockDBService;
+import ch.epfl.qedit.model.answer.AnswerModel;
 import ch.epfl.qedit.model.answer.MatrixFormat;
+import ch.epfl.qedit.model.answer.MatrixModel;
+import ch.epfl.qedit.util.Util;
 import ch.epfl.qedit.view.answer.MatrixFragment;
+import ch.epfl.qedit.viewmodel.QuizViewModel;
 import com.android21buttons.fragmenttestrule.FragmentTestRule;
+import java.util.HashMap;
 import org.junit.After;
-import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
 @RunWith(AndroidJUnit4ClassRunner.class)
 public class MatrixFragmentTest {
-    final int MATRIX_DIM = 3;
+    private final int MATRIX_DIM = 3;
+    private QuizViewModel quizViewModel;
 
     @Rule
     public final FragmentTestRule<?, MatrixFragment> testRule =
             FragmentTestRule.create(MatrixFragment.class, false, false);
 
-    @Before
```

Comment:
Why do we remove the `@Before` ? Because it forces us to explicitely call `init()` at the start of every test.

Reply:
One of the tests that I didn't finish yet actually needs a different setup. I know, not cool to remove the @Before, I'll see what I can do.

Reply:
I found a workaround by simply setting up an environment which is a compromise between both.
In that way I can test both but keep the `@Before`.

Reply:
Ok, it is fine if you need to remove the before for future tests, I was just making sure it was intentional ;).